src/utils/times.ts

```diff
@@ -0,0 +1,6 @@
+function times(n: number, func = (i: number): string | number | undefined => i): Array<number | string | undefined> {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    return Array.from({length: n}).map((_, i) => func(i));
+}
+
+export default times;
```

Comment:
```suggestion
function times<TReturnType>(n: number, func: (index: number) => TReturnType = (i) => i as TReturnType): TReturnType[] {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    return Array.from({length: n}).map((_, i) => func(i));
}

export default times;
```

The function looks nice but you are limiting it too much in the types. WDYT?